pkgs/pubspec_parse/lib/src/dependency.dart

```diff
@@ -46,10 +46,8 @@ Dependency? _fromJson(Object? data, String name) {
   }
 
   if (data is Map) {
```

Comment:
If input is expected to be JSON, then `if (data is Map<String, Object?>)` should be fine, and the cast to `String `below isn't necessary.